src/core/libmaven/Compound.h

```diff
@@ -46,147 +94,157 @@ class Compound{
+         */
+        vector<float>fragmentIntensities;
+
+        /**
+         * @brief Collection of indices of fragment values mapping to its
+         * ionisation type information.
          */
-        static string filterFormula(string formulaString)
-        {
-            string validChars = CHE_FORMULA_ALPHA_UPP
-                                + CHE_FORMULA_ALPHA_LOW
-                                + CHE_FORMULA_COFF;
-            formulaString.erase(remove_if(formulaString.begin(),
-                                          formulaString.end(),
-                                          [validChars] (const char& c) {
-                                              return (validChars.find(c)
-                                                      == string::npos);
-                                          }),
-                                formulaString.end());
-            return formulaString;
-        }
+        map<int, string>fragmentIonTypes;
 
         /**
          * @brief A simple string in form of a line notation for describing the
          * structure of chemical species using short ASCII strings.
          */
         string smileString;
 
-        // TODO: from MAVEN (upstream), find out what this is
-        string adductString;
-
-        /**
-        *@param -  srmId will hold filterLine string from mzxml file which represent type of
-        *mass spec and ionization is used and other info as well
-        */
-        string srmId;
-        float expectedRt;
+        // TODO: from MAVEN (upstream). Can this be derived somehow.
+        // Also maybe use an enum.
+        int ionizationMode;
 
-        int charge;     /**@param  -   number of charge of compound (electron loss or gain)   */
-        float mass;     /**@param   -  mass of this compund   */
-        float neutralMass;
 
-        /** QQQ mapping */
-        string method_id;   /**@param  -  TODO*/
-        float precursorMz;	/**@param  -  QQQ parent ion  mz value   */
-        float productMz;    /**@param  -  QQQ child ion   mz value */
-        float collisionEnergy; /**@param  -   QQQ collision energy of this compound   */
         float logP; // TODO: find out what value this is.
 
         // TODO: from MAVEN (upstream), find out what this is
         bool virtualFragmentation;
 
+        /**
+         * @brief A note containing any miscellaneous details for this compound.
+         */
+        string note;
+
         // TODO: from MAVEN (upstream) decoy compound?
         bool isDecoy;
 
-        // TODO: from MAVEN (upstream). Can this be derived somehow.
-        // Also maybe use an enum.
-        int ionizationMode;
-
         /**
-         * @brief Name of database this compound belongs to for example KEGG,
-         * ECOCYC, etc.
-         */
-        string db;
+        *@param  -   Reaction represents simple compound reaction
+        *Check Reaction class in mzSample.h
+        */
+
+        vector<Reaction*>reactions;
```

Comment:
Is `Compound::reactions` used anyhow?

Reply:
umm.. it's being set from database but not as such used for computations.

Reply:
Let's remove anything regarding reactions for now.

Reply:
Okkay sure